src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -136,24 +137,39 @@ def bulk_record(
             indexer_cache.set_many(new_results_to_cache, use_case_id.value)
             return cache_key_results.merge(db_read_key_results)
 
-        new_records = []
-        for write_pair in db_write_keys.as_tuples():
-            organization_id, string = write_pair
-            new_records.append(
-                self._table(use_case_id)(organization_id=int(organization_id), string=string)
-            )
+        with writes_limiter.check_write_limits(use_case_id, db_write_keys) as writes_limiter_state:
+            # After the DB has successfully committed writes, we exit this
+            # context manager and consume quotas. If the DB crashes we
+            # shouldn't consume quota.
+            filtered_db_write_keys = writes_limiter_state.accepted_keys
+            del db_write_keys
```

Comment:
is this just a safeguard against actually using this data wrong?

Reply:
yeah pretty much. it's the reason i tend to overwrite variables